pkg/services/gitops/uninstall.go

```diff
@@ -27,22 +29,42 @@ func (g *Gitops) Uninstall(params UninstallParams) error {
 
 	errorOccurred := false
 
-	if params.DryRun {
-		g.logger.Actionf("Deleting Weave Gitops manifests")
-	} else {
-		if err := g.kube.Delete(ctx, manifests.AppCRD); err != nil {
-			if !apierrors.IsNotFound(err) {
-				g.logger.Printf("received error deleting App CRD: %q", err)
+	g.logger.Actionf("Deleting Weave Gitops manifests")
 
-				errorOccurred = true
+	if !params.DryRun {
+		wegoAppManifests, err := manifests.GenerateWegoAppManifests(manifests.WegoAppParams{Version: version.Version, Namespace: params.Namespace})
+		if err != nil {
+			return fmt.Errorf("error generating wego-app manifests, %w", err)
+		}
+
+		wegoManifests := append(wegoAppManifests, manifests.AppCRD)
+		for _, m := range wegoManifests {
+			if err := g.kube.Delete(ctx, m); err != nil {
+				if !apierrors.IsNotFound(err) {
+					g.logger.Printf("error applying wego-app manifest \n%s: %w", m, err)
+
+					errorOccurred = true
+				}
 			}
 		}
 	}
 
-	if err := g.flux.Uninstall(params.Namespace, params.DryRun); err != nil {
-		g.logger.Printf("received error uninstalling flux: %q, continuing with uninstall", err)
+	wegoConfig, err := g.kube.GetWegoConfig(ctx, params.Namespace)
+	if err != nil {
+		return fmt.Errorf("failed getting wego config in namespace=%s: %w", params.Namespace, err)
+	}
+
+	uninstallFlux := true
+	if wegoConfig.FluxNamespace != params.Namespace {
+		uninstallFlux = false
```

Comment:
This confused me at first. Please add a comment that we only remove flux if it matches what our wego config is.  Or hide the flux uninstall in a function with a name like removeFluxIfMatchingWegoConfig()